### PR TITLE
Switch to docker-compose setup

### PR DESCRIPTION
**Pull request by bfbachmann**
* Add `docker-compose.yml`
* Remove `ROCKET_DOMAIN` variable from Rocket config since it's unused
* Update `dal.go` to retry when connection to PG fails
* Add `Dockerfile.db` to automatically copy SQL script to PG container

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,19 +1,22 @@
 # Start from the base Go image
 FROM golang
 
+# Set /go/src/github.com/ubclaunchpad/rocket as the CWD
+WORKDIR /go/src/github.com/ubclaunchpad/rocket
+
+# Install Postgres client to check when the DB is ready for use
+RUN apt-get update
```
[the remark this answers is not in the document]
Comment (jordanschalm) [reply Dockerfile]: I mean installing the postgres client, not the update.

File: start.sh [hunk 2]
```diff
@@ -1,24 +1,16 @@
...
 
-# Build the container from the project directory and tag as latest.
-docker build \
-    -t rocket-app:latest \
-    /go/src/github.com/ubclaunchpad/rocket
+# Wait for postgres instance to boot before starting Rocket.
 
-# Stop the old container, if it's running.
-docker stop rocket-app
+set -e
```
[the remark this answers is not in the document]
Comment (bfbachmann) [reply start.sh]: My understanding of docker-compose is that specifying `depends_on` in the compose file tells Docker to make sure the Postgres container is running before starting the Rocket container, but this doesn't guarantee that the Postgres has finished booting by the time Rocket starts. So if we don't wait for Postgres to boot before starting Rocket, Rocket errors when it starts.

[the remark this answers is not in the document]
Comment (bfbachmann) [reply start.sh]:
```
rocket_1    | time="2018-01-22T06:52:17Z" level=fatal msg="Error initializing the database: dial tcp :0: getsockopt: connection refused"
```

[the remark this answers is not in the document]
Comment (jordanschalm) [reply start.sh]: Hmm that's super interesting

[the remark this answers is not in the document]
Comment (bfbachmann) [reply start.sh]: It's a common Docker challenge https://docs.docker.com/compose/startup-order/.

File: Dockerfile [hunk 3]
```diff
@@ -1,19 +1,22 @@
...
 # Install dependencies
-RUN cd /go/src/github.com/ubclaunchpad/rocket && \
-    glide install
+RUN glide install
 
 # Build Rocket
 RUN go install github.com/ubclaunchpad/rocket
-
-# Run
-ENTRYPOINT [ "/go/bin/rocket" ]
```
[the remark this answers is not in the document]
Comment (bfbachmann) [reply Dockerfile]: So we don't have to override it to run the start script.

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  postgres:
+    build:
+      context: .
+      dockerfile: Dockerfile.db
+    ports:
+      - "5432:5432"
+    volumes:
+      - ./pgdata:/var/lib/postgresql/data
+    env_file:
+      - .db.env.example
```
Comment (jordanschalm) [inline docker-compose.yml]: I would set these to `.db.env`, so deployer just has to create the env file, not modify this file, to get started.

**Pull request merged**